bandwidth-file-spec.txt

```diff
@@ -522,3 +522,68 @@ reduced or increased as needed. Smaller quotas decrease the size
 of uncompressed consensuses, and may decrease the size of
 consensus diffs and compressed consensuses. But if the relay
 quota is too small, some relays may be over- or under-weighted.
+
+B.4. Torflow aggreation
+
+Torflow implements two methods to compute the bandwidth values from the
+(stream) bandwidth measurements: with and without PID control feedback.
+The method described here is the later (see Torflow specification,
+section 2.2).
+
+In the following sections, the relays' measured bandwidth refer to the
+ones that this bandwidth authority has measured for the relays that
+would be included in the next bandwidth authority's upcoming vote.
+
+1. Calculate the filtered bandwidth for each relay:
+   - choose the relay's measurements that are equal or greater than the
+     mean of the measurements for this relay
+   - calculate the mean of those measurements
+
+   In pseudocode:
+
+      bw_filt_i = mean(max(mean(bw_j), bw_j))
+
+2. Calculate network averages:
+   - calculate the filtered average by dividing the sum of all the
+     relays' filtered bandwidth by the number of relays that have been
+     measured, ie, calculate the mean average of the relays' filtered
+     bandwidth.
+   - calculate the stream average by dividing the sum of all the
+     relays' filtered bandwidth by the number of relays that have been
+     measured, ie, calculate the mean average or the relays' measured
+     bandwidth.
+
+    In pseudocode:
+
```

Comment:
n = sum(bw_i)
    n_filt = sum(bw_filt_i)

Reply:
hmm, i wanted to try if i can also create suggestion code, but it seems not

Reply:
n_filt is unused